### PR TITLE
lzfse: modernize derivation

### PR DESCRIPTION
**Pull request by KSJ2000**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by HeitorAugustoLN — COMMENTED** [review]
It would be better if each change would be made in a separate commit here, such as nixfmt, removing `with lib`, etc.

**Comment by HeitorAugustoLN** [comment]
> > It would be better if each change would be made in a separate commit here, such as nixfmt, removing `with lib`, etc.
> 
> nixfmt? I did no such thing. My alphabetization just happened to coincide; the package was already fmted. Please point me to where it says `with lib` should be a separate commit, because by that format, replacing rec with finalAttrs should be a separate commit as well.

I thought you had formatted. But not necessarily it needs to be a separate commit, but it would be better if the commit explained everything that the commit does, and `replace sha256 with hash` is not the only thing being done. So it could have a message such as `modernize derivation` or something like that

https://github.com/NixOS/nixpkgs/blob/73bd40281687e5bd0dadb63b34865231fd20e7e4/CONTRIBUTING.md?plain=1#L497

File: pkgs/by-name/lz/lzfse/package.nix [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
+stdenv.mkDerivation (finalAttrs: {
   pname = "lzfse";
   version = "1.0";
 
   src = fetchFromGitHub {
     owner = "lzfse";
     repo = "lzfse";
-    rev = "lzfse-${version}";
-    sha256 = "1mfh6y6vpvxsdwmqmfbkqkwvxc0pz2dqqc72c6fk9sbsrxxaghd5";
+    rev = "88e2d2788b4021d0b2eb9fe2d97352ae9190f128";
```
Comment (Aleksanaa) [inline pkgs/by-name/lz/lzfse/package.nix:15]:
```suggestion
    rev = "lzfse-${finalAttrs.version}";
```

Comment (Aleksanaa) [inline pkgs/by-name/lz/lzfse/package.nix:15]:
```suggestion
    rev = "lzfse-${finalAttrs.version}";
```

**Review by FliegendeWurst — APPROVED** [review]
No rebuilds :heavy_check_mark: .

**Review by HeitorAugustoLN — APPROVED** [review]

File: pkgs/by-name/lz/lzfse/package.nix [hunk 2]
```diff
@@ -1,29 +1,34 @@
...
       This is a reference C implementation of the LZFSE compressor introduced in the Compression library with OS X 10.11 and iOS 9.
       LZFSE is a Lempel-Ziv style data compression algorithm using Finite State Entropy coding.
       It targets similar compression rates at higher compression and decompression speed compared to deflate using zlib.
     '';
-    platforms = platforms.unix;
-    license = licenses.bsd3;
-    maintainers = [ ];
+    platforms = lib.platforms.unix;
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ KSJ2000 ];
```
Comment (tomodachi94) [inline pkgs/by-name/lz/lzfse/package.nix:31]: You're missing from `maintainers/maintainer-list.nix`. Please add yourself ([instructions](https://github.com/NixOS/nixpkgs/tree/master/maintainers#how-to-become-a-maintainer)), taking care to use a separate commit. I encourage you to use this same PR to add yourself to the maintainers list.

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Thank you for adopting this package! Most of the diff looks okay, except you're missing from the maintainers list (see comment).

**Comment by Pandapip1** [comment]
Merge conflict due to https://github.com/NixOS/nixpkgs/pull/354531.

File: pkgs/by-name/lz/lzfse/package.nix (hunk 2 above)
Comment (tomodachi94) [reply pkgs/by-name/lz/lzfse/package.nix:31]: @KSJ2000 please see above :)

Comment (tomodachi94) [reply pkgs/by-name/lz/lzfse/package.nix:31]: Added in https://github.com/NixOS/nixpkgs/pull/348979, waiting on [a rebase](https://github.com/NixOS/nixpkgs/pull/348979#issuecomment-2475643855)

**Review by Luflosi — APPROVED** [review]
LGTM.

**Review by HeitorAugustoLN — APPROVED** [review]

**Review by tomodachi94 — APPROVED** [review]
Lgtm, sorry for the wait ♥️

**Pull request merged**